### PR TITLE
Show Filter Label, `display` for Field Choices for Active Filter Display

### PR DESCRIPTION
**Pull request by bryanculver**
# Closes: #2989
# What's Changed
- Attempt to show a label for a filter field instead of the filter name.
- Attempt to show a user-friendly display of the filter choice.

This:
image

instead of:
image

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [NA] Documentation Updates (when adding/changing features)
- [NA] Example Plugin Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/core/views/generic.py [hunk 1]
```diff
@@ -252,9 +255,25 @@ def get(self, request):
...
+
+                label = filterset_filters[field_name].label
+                if len(values) == 0 or not is_uuid(values[0]):
+                    return [label, values]
+                else:
+                    try:
+                        filtered_results = filterset_filters[field_name].queryset.filter(pk__in=values)
+                        new_values = [result.display for result in filtered_results]
+                        return [label, new_values]
+                    except (FieldError, AttributeError):
```
Comment (HanlinMiao) [inline nautobot/core/views/generic.py:272]: Just curious, which statement could cause a FieldError?

File: nautobot/core/views/generic.py [hunk 2]
```diff
@@ -252,9 +255,25 @@ def get(self, request):
                 )
                 self.queryset = self.queryset.none()
 
+            def check_for_display(field_name, values):
```
Comment (HanlinMiao) [inline nautobot/core/views/generic.py:258]: Shall we add one for `NautobotUIViewSet's` mixin as well? Maybe this should be a helper method in `utilities` so both places can use it without much redundant code.

**Review by HanlinMiao — CHANGES_REQUESTED** [review]

File: nautobot/core/views/generic.py (hunk 1 above)
Comment (bryanculver) [reply nautobot/core/views/generic.py:272]: `result.display`, just in case the object doesn't have a `display` property

Comment (HanlinMiao) [reply nautobot/core/views/generic.py:272]: Ahhh i thought that was `AttributeError`

**Review by HanlinMiao — APPROVED** [review]
Love the super comprehensive test case!

**Pull request merged**